File: Scadex.Model/Dtos/Diagram/Queries/Items/DiagramCanvasSettingsDto.cs
using Scadex.Core.Model;
using static Scadex.Model.Enums.EntityEnums;

namespace Scadex.Model.Dtos.Diagram.Queries.Items;

public class DiagramCanvasSettingsDto : IDto
{
    public int GridSize { get; set; }
    public bool SnapToGrid { get; set; }
    public BackgroundVariant BackgroundVariant { get; set; }
    public string GridColor { get; set; } = null!;
    public string BackgroundColor { get; set; } = null!;
    public double MinZoom { get; set; }
    public double MaxZoom { get; set; }
}

File: Scadex.WebUI/src/models/diagram/queries/diagramCanvasSettingsDto.ts
/** Ayna: Scadex.Model/Dtos/Diagram/Queries/DiagramCanvasSettingsDto.cs — sözleşme: docs/api-contract/02-diagram-read.md */
import type { BackgroundVariant } from '@/models/enums';

/**
 * Kabinin canvas tercihleri. Aggregate'in İÇİNDE gelir — ayrı çağrı yoktur.
 *
 * `id` ve audit alanları BİLEREK yok: kabinin kayıtlı ayarı yoksa sunucu
 * varsayılan döner ve satır oluşturmaz, o yanıtta bir kimlik olamaz.
 */
export interface DiagramCanvasSettingsDto {
  gridSize: number;
  snapToGrid: boolean;
  backgroundVariant: BackgroundVariant;
  gridColor: string;
  backgroundColor: string;
  minZoom: number;
  maxZoom: number;
}
